### PR TITLE
Attribute unit tests

### PR DESCRIPTION
**Pull request by fmorency**

### CODE REVIEW

**Comment by fmorency** [comment]
Sorry for poking you and converting this to a draft. I'm discovering `proptest` and it gave me some ideas.

**Comment by fmorency** [comment]
@hansl Please review this PR.

File: src/many/src/protocol.rs [hunk 1]
```diff
@@ -91,27 +64,19 @@ impl Ord for Attribute {
...
     fn encode<W: Write>(&self, e: &mut Encoder<W>) -> Result<(), Error<W::Error>> {
-        if let Some(arguments) = self.arguments() {
-            e.array(1 + arguments.len() as u64)?;
-            e.u32(self.id as u32)?;
-            for a in arguments {
-                e.encode(a)?;
-            }
-        } else {
-            e.u32(self.id as u32)?;
+        e.array(1 + self.arguments.len() as u64)?;
```
Comment (hansl) [inline src/many/src/protocol.rs:76]: If there's no argument, the encoding is still just the attribute number.

Comment (fmorency) [reply src/many/src/protocol.rs:76]: Good catch! I will add a test.

File: src/many/src/protocol.rs [hunk 2]
```diff
@@ -13,59 +13,32 @@ pub type AttributeId = u32;
 #[derive(Clone)]
```
Comment (hansl) [inline src/many/src/protocol.rs:13]:
```suggestion
#[derive(Clone, Debug)]
```

File: src/many/src/protocol.rs [hunk 3]
```diff
@@ -91,27 +64,23 @@ impl Ord for Attribute {
 
 impl Debug for Attribute {
```
Comment (hansl) [inline src/many/src/protocol.rs:65]: And remove this.

**Review by hansl — APPROVED** [review]

**Pull request merged**